include/envoy/server/filter_config.h

```diff
@@ -90,11 +90,12 @@ class ProtocolOptionsFactory : public Config::TypedFactory {
    * @param validation_visitor message validation visitor instance.
    * @return Upstream::ProtocolOptionsConfigConstSharedPtr the protocol options
    */
-  virtual Upstream::ProtocolOptionsConfigConstSharedPtr
-  createProtocolOptionsConfig(const Protobuf::Message& config,
-                              ProtobufMessage::ValidationVisitor& validation_visitor) {
+  virtual Upstream::ProtocolOptionsConfigConstSharedPtr createProtocolOptionsConfig(
+      const Protobuf::Message& config, ProtobufMessage::ValidationVisitor& validation_visitor,
+      Server::Configuration::TransportSocketFactoryContext& factory_context) {
     UNREFERENCED_PARAMETER(config);
     UNREFERENCED_PARAMETER(validation_visitor);
+    UNREFERENCED_PARAMETER(factory_context);
```

Comment:
nit: this implementation does not obey the spec, it should through an exception instead.

Reply:
the implementation seems to be ok with returned nullptr: 
https://github.com/envoyproxy/envoy/blob/68d74ade909e1eb757da674f10ba73ccad7f93c2/source/common/upstream/upstream_impl.cc#L191

we should either change the comment about the exception, or add a RELEASE_ASSERT near that line to ensure that nullptrs are NACK'ed..

any preference?

Reply:
actually on a second look, the nill check is needed to support both config and typed_config.
I'll just add an exception to the method as you suggested

Reply:
for now added the exception. will change if needed